monkey/monkey_island/cc/resources/agent_events.py

```diff
@@ -106,20 +133,32 @@ def _parse_event_filter_args(
                     "expected timestamp to be a number"
                 )
 
-        return type_, success, timestamp_constraint
+        return timestamp_constraint
 
     def _get_filtered_events(
         self,
         type_: Optional[Type[AbstractAgentEvent]],
+        tag: Optional[str],
         success: Optional[bool],
         timestamp_constraint: Optional[Tuple[str, float]],
     ) -> Sequence[AbstractAgentEvent]:
         if type_ is not None:
-            events: Sequence[AbstractAgentEvent] = self._agent_event_repository.get_events_by_type(
-                type_
-            )
+            events_by_type: Sequence[
+                AbstractAgentEvent
+            ] = self._agent_event_repository.get_events_by_type(type_)
         else:
-            events = self._agent_event_repository.get_events()
+            events_by_type = self._agent_event_repository.get_events()
+
+        if tag is not None:
+            events_by_tag: Sequence[
+                AbstractAgentEvent
+            ] = self._agent_event_repository.get_events_by_tag(tag)
```

Comment:
A minor performance improvement might be to extract this as a method and just say `return events_by_type`

Reply:
I don't think I understand. Extract what exactly?